File: 04.Simulation模拟/leetcode_2264_largestGoodInteger.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        n = len(num)
        res = -1
        for i in range(1, n-1):
            if num[i-1] == num[i] and num[i] == num[i+1]:
                if res<int(num[i]):
                    res = int(num[i])
        return 3*str(res) if res!=-1 else ""

if __name__ == "__main__":
    s = Solution()
    num = "6777133339"
    print(s.largestGoodInteger(num))

File: 04.Simulation模拟/leetcode_2264_largestGoodInteger.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    string largestGoodInteger(string num) {
        int n = num.size();
        int res = -1;
        for (int i=1; i<n-1; i++){
            if (num[i-1] == num[i] && num[i] == num[i+1]){
                res = max(res, num[i]-'0');
            }
        }
        string ans(3, '0'+res);
        return res==-1 ? "" : ans;
    }
};

int main(){
    Solution s;
    string num = "6777133339";
    cout << s.largestGoodInteger(num) << endl;
}
